app.go

```diff
@@ -1024,7 +1033,66 @@ func (app *App) startupProcess() *App {
 	}
 
 	app.mutex.Lock()
+	defer app.mutex.Unlock()
+
+	// add routes of sub-apps
+	if !app.subAppsRoutesAdded {
+		app.appendSubAppLists(app.appList)
+		app.addSubAppsRoutes(app.appList)
+
+		app.subAppsRoutesAdded = true
+	}
+
+	// build route tree stack
 	app.buildTree()
-	app.mutex.Unlock()
+
 	return app
 }
+
+// appendSubAppLists supports nested for sub apps
+func (app *App) appendSubAppLists(appList map[string]*App, parent ...string) {
+	for prefix, subApp := range appList {
+		// skip real app
+		if prefix == "" {
+			continue
+		}
+
+		if len(parent) > 0 {
+			prefix = parent[0] + prefix
+		}
+
+		if _, ok := app.appList[prefix]; !ok {
+			app.appList[prefix] = subApp
+		}
+
+		if len(subApp.appList) > 1 {
```

Comment:
please add a comment here "why it should be greater than 1"

i know its because of the real app -> but the next dev should also know this